File: cpe-level-1/python/zj-d260.py
#!/usr/bin/env python 

T = int(input())
for _ in range(T):
    a = list(map(int, input().split()))
    a.sort()
    if a[0] == a[3]:
        print("square")
    elif a[0] == a[1] and a[2] == a[3]:
        print("rectangle")
    else:
        if a[0]+a[1]+a[2] > a[3]:
            print("quadrangle")
        else:
            print("banana")

File: cpe-level-1/cpp/zj-d260.cpp

#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int T;
    int a[4];
    cin >> T;
    while (T--){
        for (int i = 0; i < 4; i++){
            cin >> a[i];
        }
        sort(a, a+4);
        if (a[0] == a[3]) cout << "square\n";
        else if (a[0] == a[1] && a[2] == a[3]) cout << "rectangle\n";
        else {
            if (a[0] + a[1] + a[2] > a[3]) cout << "quadrangle\n";
            else cout << "banana\n";
        }
    }
    return 0;
}
